File: algo/src/test/java/com/hartwig/actin/algo/evaluation/general/IsLegallyInstitutionalizedTest.java
package com.hartwig.actin.algo.evaluation.general;

import static com.hartwig.actin.algo.evaluation.EvaluationAssert.assertEvaluation;

import com.hartwig.actin.TestDataFactory;
import com.hartwig.actin.algo.datamodel.EvaluationResult;

import org.junit.Test;

public class IsLegallyInstitutionalizedTest {

    @Test
    public void canEvaluate() {
        IsLegallyInstitutionalized function = new IsLegallyInstitutionalized();

        assertEvaluation(EvaluationResult.NOT_EVALUATED, function.evaluate(TestDataFactory.createMinimalTestPatientRecord()));
    }
}

File: algo/src/test/kotlin/com/hartwig/actin/algo/evaluation/general/IsLegallyInstitutionalizedTest.kt
package com.hartwig.actin.algo.evaluation.general

import com.hartwig.actin.algo.evaluation.EvaluationAssert
import com.hartwig.actin.algo.evaluation.EvaluationLabels
import com.hartwig.actin.configuration.ReportIntendedUse
import com.hartwig.actin.datamodel.TestPatientFactory
import com.hartwig.actin.datamodel.algo.EvaluationResult
import org.junit.jupiter.api.Test

class IsLegallyInstitutionalizedTest {

    private val function = IsLegallyInstitutionalized(EvaluationLabels.load(ReportIntendedUse.RESEARCH_USE_ONLY).general)

    @Test
    fun `Should fail always`() {
        EvaluationAssert.assertEvaluation(
            EvaluationResult.FAIL, function.evaluate(TestPatientFactory.createMinimalTestWGSPatientRecord())
        )
    }
}